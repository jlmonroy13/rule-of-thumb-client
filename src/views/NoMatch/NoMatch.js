import React from 'react';
import { NavLink } from 'react-router-dom';

const NoMatch = () => (
  <div className="d-flex justify-content-center align-items-center w-100 vh-100">
    <div className="text-center">
      <h1>Page not found</h1>
      <p>The page you were looking for was moved or doesnâ€™t exist.</p>
      <NavLink to="/">Go to Homepage</NavLink>
    </div>
  </div>
);

export default NoMatch;
