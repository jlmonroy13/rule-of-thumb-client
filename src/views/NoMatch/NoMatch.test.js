/**
 *
 * Tests for NoMatch
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import NoMatch from './index';

const props = {};

describe('<NoMatch />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Router>
        <NoMatch {...props} />
      </Router>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Router>
        <NoMatch {...props} />
      </Router>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
