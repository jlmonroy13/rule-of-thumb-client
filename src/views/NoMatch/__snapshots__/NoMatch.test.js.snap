// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoMatch /> Should render and match the snapshot 1`] = `
<div
  class="d-flex justify-content-center align-items-center w-100 vh-100"
>
  <div
    class="text-center"
  >
    <h1>
      Page not found
    </h1>
    <p>
      The page you were looking for was moved or doesnâ€™t exist.
    </p>
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      Go to Homepage
    </a>
  </div>
</div>
`;
